packages/cli/lib/services/response-saver.service.ts

```diff
@@ -1,38 +1,35 @@
 import fs from 'node:fs';
 import path from 'node:path';
-import type { AxiosResponse } from 'axios';
+import crypto from 'node:crypto';
+import type { AxiosRequestConfig, AxiosResponse } from 'axios';
 import type { Connection } from '@nangohq/shared';
 import type { Metadata } from '@nangohq/types';
 
+const FILTER_HEADERS = ['authorization', 'user-agent', 'nango-proxy-user-agent', 'accept-encoding', 'retries', 'host'];
+
+interface CachedRequest {
+    requestIdentityHash: string;
+    requestIdentity: unknown[];
```

Comment:
why `unknown[]`? It looks like requestIdentity is an object of type `{ method, url, params, headers?, data? }`

Reply:
Check out around line 94. RequestIdentity is an array of key,value tuples so that we can sort it by key and always get a consistent hash.

Reply:
cannot we not get a consistent hash with a object?

Reply:
It's insertion order so if someone were to for some reason change the key order in their code it would also invalidate the data.

Reply:
I get the ordering of params and headers and those should be stored in sorted lists to ensure comparison but I don't understand why the high level attributes are stored in an array themselves. 
Hashing a `{ method: string, url: string, params: [string, unknown][], headers: [string, unknown][]}`object would also produce a unique hash

Reply:
Ah yeah, I see now. I'll clean that up shortly.

Reply:
Fixed in 9f30ffd4